fs/fs.go

```diff
@@ -0,0 +1,517 @@
+	stat.Nlink = node.stat.Nlink
+	stat.Uid = node.stat.Uid
+	stat.Gid = node.stat.Gid
+	stat.Rdev = node.stat.Rdev
+	stat.Atim = node.stat.Atim
+	stat.Mtim = node.stat.Mtim
+	stat.Ctim = node.stat.Ctim
+	stat.Blksize = node.stat.Blksize
+	stat.Blocks = node.stat.Blocks
+	stat.Birthtim = node.stat.Birthtim
+	stat.Flags = node.stat.Flags
+
+	return 0
+}
+
+func (fs *UorFs) Read(path string, buff []byte, ofst int64, fh uint64) (n int) {
+	defer fs.synchronize()()
+
+	node := fs.lookupNode(path)
+	if node == nil {
+		return -fuse.ENOENT
+	}
+
+	if node.data == nil {
+		nodeData, err := fs.client.GetContent(fs.ctx, fs.Source, *node.desc)
+		if err != nil {
+			return -fuse.ENOENT
+		}
+
+		node.data = NewDecayCache(&nodeData, fs.cacheDuration, &fs.Logger)
+	}
+	node.data.AddUser()
+	defer node.data.RemoveUser()
+
+	endofst := ofst + int64(len(buff))
+	if endofst > int64(len(*node.data.data)) {
+		endofst = int64(len(*node.data.data))
+	}
+	if endofst < ofst {
+		return 0
+	}
+	n = copy(buff, (*node.data.data)[ofst:endofst])
+	return
+}
+
+func (fs *UorFs) Readdir(path string, fill func(name string, stat *fuse.Stat_t, ofst int64) bool, ofst int64, fh uint64) (errc int) {
+	defer fs.synchronize()()
+	fill(".", nil, 0)
+	fill("..", nil, 0)
+
+	node := fs.lookupNode(path)
+	if node == nil {
+		return -fuse.ENOENT
+	}
+
+	for name, child := range node.children {
+		if !fill(name, &child.stat, 0) {
+			break
+		}
+	}
+	return 0
+
+}
+
+func (fs *UorFs) Listxattr(path string, fill func(name string) bool) (errc int) {
+	defer fs.synchronize()()
+	node := fs.lookupNode(path)
+	if node == nil {
+		return -fuse.ENOENT
+	}
+	for name := range node.xattrs {
+		if !fill(name) {
+			return -fuse.ERANGE
+		}
+	}
+	return 0
+}
+
+func (fs *UorFs) Getxattr(path string, name string) (errc int, xattr []byte) {
+	defer fs.synchronize()()
+	node := fs.lookupNode(path)
+	if node == nil {
+		return -fuse.ENOENT, nil
+	}
+	if xattr, ok := node.xattrs[name]; !ok {
+		return -fuse.ENOATTR, nil
+	} else {
+		return 0, xattr
+	}
+
+}
+
+func (fs *UorFs) synchronize() func() {
+	fs.mutex.Lock()
+	return func() {
+		fs.mutex.Unlock()
+	}
+}
+
+// loadFromReference loads a collection from an image reference.
+func (fs *UorFs) loadFromReference(ctx context.Context, reference string, client registryclient.Remote) error {
+
+	layerDescriptors, err := getManifest(ctx, reference, client, fs.matcher)
+	if err != nil {
+		return err
+	}
+
+	for _, layerInfo := range layerDescriptors {
+		layerInfo := layerInfo // fix &layerInfo
+
+		switch layerInfo.MediaType {
+		case ocimanifest.UORSchemaMediaType:
+			continue
+		case ocispec.MediaTypeImageConfig:
+			continue
+		case ocimanifest.UORConfigMediaType:
+			continue
+		case ocispec.MediaTypeImageManifest:
+			continue
+		}
+
+		if layerInfo.Annotations == nil {
+			fs.Logger.Debugf("layer unexpectedly had no annotations, ignoring: %v", layerInfo.Digest)
+			continue
+		}
+
+		skip := func(_ string) bool { return false }
```

Comment:
Nit: You can pass `nil` here. The function it calls checks if the `skip` function is nil before it's evaluated.